templates/problem/voting-stats.html

```diff
@@ -0,0 +1,383 @@
+            ctx.strokeStyle = 'rgb(94,101,105)';
+            ctx.lineWidth = 1;
+            let vertical_line_start_y = plot_padding;
+            let vertical_line_end_y = plot_padding + plot_height;
+            for (let i = 0; i <= 10; i++) {
+                line(points_to_x(points), vertical_line_start_y, points_to_x(points), vertical_line_end_y);
+                ctx.fillText(points.toFixed(2) + 'p', points_to_x(points), points_text_y);
+                points += points_range / 10.0;
+            }
+            //draw the horizontal lines
+            ctx.strokeStyle = 'rgb(94,101,105)';
+            let horizontal_line_start_x = plot_padding;
+            let horizontal_line_end_x = plot_padding + plot_width;
+            let horizontal_line_y = plot_padding;
+            for (let i = 0; i <= 4; i++) {
+                line(horizontal_line_start_x, horizontal_line_y, horizontal_line_end_x, horizontal_line_y)
+                horizontal_line_y += plot_height / 4.0;
+            }
+            //get the animated x coordinates of a given x based on current frame
+            function get_animated_x(final_x) {
+                let initial_x = plot_padding + plot_width / 2.0;
+                return initial_x + frames * ((final_x - initial_x) / total_frames);
+            }
+
+            function points_to_animated_x(points) {
+                return get_animated_x(points_to_x(points));
+            }
+
+            {% if enough_data_for_plot %} //render box plot
+                if (document.getElementById('id_show_box_plot_box').checked) {
+                    //box and whisker plot settings
+                    const box_plot_height = plot_height * (3 / 4);
+                    const box_plot_box_y = plot_padding+plot_height/2-box_plot_height/2;
+                    const box_plot_edges_height = plot_height/2;
+                    const box_plot_edges_y = plot_padding+plot_height/2-box_plot_edges_height/2;
+                    const box_plot_middle_y = plot_padding+plot_height/2;
+
+                    //box and whisker plot data
+                    const box_plot_data = [{{min_vote}},{{first_quartile}},{{median_vote}},{{third_quartile}},{{max_vote}}];
+                    const render_box_plot = [true,true,true,true,true];
+                    for(let i=0;i<2;i++) if(box_plot_data[i] === box_plot_data[i+1]) render_box_plot[i] = false;
+                    for(let i=3;i<5;i++) if(box_plot_data[i] === box_plot_data[i-1]) render_box_plot[i] = false;
+
+                    //turning this garbage into for loop, WIP
+                    const box_plot_x = [];
+                    for (let i = 0; i < 5; i++) box_plot_x.push(points_to_animated_x(box_plot_data[i]));
+                    const box_plot_y = [box_plot_edges_y, box_plot_box_y, -1, box_plot_box_y, box_plot_edges_y]
+
+                    ctx.strokeStyle = '#000000';
+                    ctx.lineWidth = 5;
+                    for (let i = 0; i < 5; i++){
+
+                    }
```

Comment:
Delete.

Reply:
this is WIP for now, I'm still working on this.